src/main/java/com/ldtteam/structurize/network/NetworkChannel.java

```diff
@@ -11,25 +17,53 @@
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.chunk.LevelChunk;
 import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.ModList;
 import net.minecraftforge.network.NetworkEvent;
 import net.minecraftforge.network.NetworkRegistry;
 import net.minecraftforge.network.PacketDistributor;
 import net.minecraftforge.network.simple.SimpleChannel;
 
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
 import java.util.function.Function;
 
 /**
  * Our wrapper for Forge network layer
  */
 public class NetworkChannel
 {
-    private static final String LATEST_PROTO_VER = "1.1";
-    private static final String        ACCEPTED_PROTO_VERS = LATEST_PROTO_VER;
     /**
      * Forge network channel
      */
     private final        SimpleChannel rawChannel;
 
+    /**
+     * The messages that this channel can process, as viewed from a message id.
+     */
+    private final Map<Integer, NetworkingMessageEntry<?>> messagesTypes = Maps.newHashMap();
+
+    /**
+     * The message that this channel can process, as viewed from a message type.
+     */
+    private final Map<Class<? extends IMessage>, Integer> messageTypeToIdMap = Maps.newHashMap();
+
+    /**
+     * Cache of partially received messages, this holds the data untill it is processed.
+     */
+    private final Cache<Integer, Map<Integer, byte[]>> messageCache = CacheBuilder.newBuilder()
```

Comment:
If you want you might want to fix this here, since on the server side this could collide if players send two packets with the same com id.